tests/inlineasm/xtensa/asmloadstore.py

```diff
@@ -0,0 +1,93 @@
+import array
+
+# On the 8266 the generated code gets put into the IRAM segment, which is only
+# word-addressable.  Therefore, to test byte and halfword load/store opcodes
+# some memory must be reserved in the DRAM segment.
+
+BYTE_DATA = array.array("B", (0x11, 0x22, 0x33, 0x44))
+WORD_DATA = array.array("h", (100, 200, -100, -200))
+DWORD_DATA = array.array("i", (100_000, -200_000, 300_000, -400_000))
+
+
+@micropython.asm_xtensa
+def tl32r() -> int:
+    j(CODE)
+    label(DATA)
+    data(1, 1, 2, 3, 4, 5, 6, 7)
+    label(CODE)
+    l32r(a2, DATA)
```

Comment:
This test fails the alignment requirements for `l32r`.

If you insert 3x nop's after `data()` then it works.

But then if you insert a nop after the `j(CODE)` it gives different answers!

Reply:
This is weird, because I recall this working back then.  It might have been some changes that were probably stashed and then lost when I re-cloned the repo.

Anyway, I'll look into it today.

Reply:
OK, the `l32r` test should be sorted out.  The code was pretty much correct, but it depends on the both the target address and the relative distance being word-aligned.

Unless I missed something, labels in inline asm blocks do not have a default alignment, so additional `align(4)` directives had to be strategically placed to make things work.  I wonder if there's a way to easily override the behaviour of `label()` just for xtensa, so those alignment directives can be made implicit.

Reply:
I feel like it should be up to the user to know about alignment in this particular case.  Because if all labels are automatically aligned (and padded with `nop_n`?) then that could very well be unexpected to the user.  And alignment is only necessary for l32r, right?

Probably documentation is the best way to address this.  And good error messages, see above about emitting a different error for out-of-alignment.

Reply:
From what I can see, word-alignment is only necessary for `l32r` and `call0`-like opcodes.  Looking for `32-bit aligned` in Cadence's ISA documentation brings up only those opcodes.

I agree on the out of alignment message, I'll make the necessary changes to have something like `ERROR: label "%q" %d bytes out of alignment` in those cases.